### ISSUE TITLE
org-journal-previous-entry always goes to first entry

### ISSUE THREAD

**Issue by sergiusens** [bastibe/org-journal#295]
**Describe the bug**
Seem that since the merge of https://github.com/bastibe/org-journal/commit/7f96657df3875b9b3633fb903e7cdaa9f1d9e117 org-journal-previous-entry always takes me to the first journal entry I have

**To Reproduce**
- Use Daily journal file
- Locate a recent entry from many
- call org-journal-previous-entry

**Expected behavior**
I expect going to the previous journal entry

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04 (KDE Neon)
 - Emacs Version 27.1

**Your Emacs Configuration**

```
  (use-package org-journal
    :ensure t
    :defer t
    :bind
    ("C-c j j" . org-journal-new-entry)
    :custom
    (org-journal-dir "~/org/journal/")
    (org-journal-enable-agenda-integration t)
    (org-journal-file-format "%Y-%m-%d.org")
    :init
    ;; Prefijo para cargar org-journal
    (setq org-journal-prefix-key "C-c j ")
    :config
    (defun ss/org-journal-file-header-func (time)
      "Custom function to create journal header."
      (concat
       (pcase org-journal-file-type
         (`daily "#+TITLE: Agenda Diaria\n#+STARTUP: content")
         (`weekly "#+TITLE: Agenda Semanal\n#+STARTUP: content")
         (`monthly "#+TITLE: Agenda Mensual\n#+STARTUP: content")
         (`yearly "#+TITLE: Agenda Anual\n#+STARTUP: content"))))

    (defun ss/new-buffer-p ()
      (not (file-exists-p (buffer-file-name))))

    (defun ss/org-journal-daily ()
      (let ((template-file (expand-file-name "agenda-diaria.org" ss/org-template-directory)))
        (when (ss/new-buffer-p)
          (save-excursion
            (goto-char (point-max))
            (insert-file-contents template-file)))))

    (add-hook 'org-journal-after-header-create-hook #'ss/org-journal-daily)

    (setq org-journal-file-header 'ss/org-journal-file-header-func
	      org-journal-carryover-items "TODO=\"TODO\"|TODO=\"WAIT\"|TODO=\"HOLD\""))
```

**Comment by wudanyang6** [comment]
![image](https://user-images.githubusercontent.com/8189540/92955378-99745b00-f497-11ea-9ec1-8158a71640bf.png)

I debug on this package, and comment unless line to force sort dates then it works fine.

**Comment by cslux** [comment]
Fixed via e7c51c4. Thanks for the report!
